src/Application/Commands/SendStatistics.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Commands;
+
+use Aws\CloudWatch\CloudWatchClient;
+use MatchBot\Domain\DonationRepository;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Send CloudWatch Metrics values of:
+ * * overall completion rate of donations from ~16 minutes old thru ~1 minutes old, if 20+
+ *   such donations started in that time
+ * * donations created (in previous full minute before task start)
+ * * donations collected (in previous full minute before task start)
+ */
+#[AsCommand(
+    name: 'matchbot:send-statistics',
+    description: 'Sends CloudWatch headline figures on donation initiation and collection'
+)]
+class SendStatistics extends LockingCommand
+{
+    public function __construct(
+        private CloudWatchClient $cloudWatchClient,
+        private DonationRepository $donationRepository
+    ) {
+        parent::__construct();
+    }
+
+    protected function doExecute(InputInterface $input, OutputInterface $output): int
+    {
+        $time = time();
+        $startOfThisMinute = new \DateTimeImmutable('@' . ($time - ($time % 60)));
```

Comment:
The logic for finding the start of the minute is effectively untested, since its repeated in the test class and just compared to itself. I think it probably is worth injecting either a clock or a timestamp so we can use fixed values in the test to see what the result of this calculation is. A timestamp would be enough, if you want to do a clock we'd use it like this:

```suggestion
        $time = $this->clock->now()->getTimestamp();
        $startOfThisMinute = new \DateTimeImmutable('@' . ($time - ($time % 60)));
```

Reply:
Added testing against a specific time in 4bda1919d6407cc86ef89efec3cc6f1f1da62d67